### ISSUE TITLE
Next arrow on last slide causes the previous slide to transition on top of it

### ISSUE THREAD

**Issue by kamui** [xiaolin/react-image-gallery#173]
This regression was introduced in 0.7.15 and doesn't exist in 0.7.14 and occurs in Safari, works fine in Chrome.

If you have 3 slides, infinite scroll is on, and you click to the last slide, clicking the next arrow causes the previous slide to slide on top of the current slide to the right.

It used to slide behind, so you couldn't see it being moved.

Here's a clip of it occurring:

https://www.dropbox.com/s/h7f90vrjzrjr6xm/react-image-gallery_bug.mov?dl=0

Dropbox's video player compresses the video so it's hard to catch the animation. I'd recommend downloading it and playing it.

**Comment by ajwagner777** [comment]
Similar issue here.  I am using a custom renderer to make text-based slides.

I slowed down the animation to see it in action.  

Here's my code:

```
<ImageGallery
  items={[
    { quote: "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Harum sed in molestias aspernatur quae quis error ipsam sunt totam ab autem, eveniet nihil, enim nobis eligendi soluta recusandae sapiente." },
    { quote: "Consectetur adipisicing elit. Harum sed in molestias aspernatur quae quis error ipsam sunt totam ab autem, eveniet nihil, enim nobis eligendi perferendis soluta recusandae sapiente." },
    { quote: "Harum sed in molestias aspernatur quae quis error error consectetur adipisicing elit ipsam sunt totam ab autem, eveniet nihil, enim nobis eligendi perferendis soluta recusandae sapiente." }
  ]}
  showThumbnails={false}
  showFullscreenButton={false}
  showPlayButton={false}
  renderLeftNav={(onClick, disabled) => navRenderer("prev", onClick, disabled)}
  renderRightNav={(onClick, disabled) => navRenderer("next", onClick, disabled)}
  renderItem={item => testimonialRenderer(item)} />
```

testimonialRenderer: 

```
testimonialRenderer = (item) => {
  return <blockquote>
    {item.quote}<br/>
    <Link to="" className="view-all">View all customer testimonials</Link>
  </blockquote>
}
```

Gonna make a stab at getting it fixed.  I suspect it's z-index related (it always is... ).

**Comment by xiaolin** [comment]
If its working a patch ago then probably something that can be tracked by commit history. I can take a look(if there are no prs) when I get back next week.

**Comment by ajwagner777** [comment]
It's not a perfect solution, but it is passable enough for the time being (sorry, don't have time for a PR... deadlines... ):

In your CSS, override the z-index for the left, right and center classes.  Like so (in sass):

```
.image-gallery-slide {
  height: 450px;
  background: $gray-lighter;

  &.left {
    z-index: 20 !important; // I feel dirty using !importants... feels like giving up
  }

  &.center {
    z-index: 30 !important;
  }

  &.right {
    z-index: 10 !important;
  }
}
```

**Comment by xiaolin** [comment]
@kamui I could not reproduce this using the `example/app.js` file in the project, any chance there might be a css collision?

**Comment by kamui** [comment]
I'll try to replicate it again.

**Comment by cyresseguier** [comment]
Similar issue as above. Visible on Chrome last version or Safari last version

**Comment by kevinlaw91** [comment]
Same issue in FF53 and Edge 40. Works correctly in Chrome.

**Comment by xiaolin** [comment]
I'm able to see a weird flicker on ff53, very odd.

**Comment by xiaolin** [comment]
This is now fixed in the latest release `v0.8.1`
